src/models/DefaultPluginData.js

```diff
@@ -0,0 +1,91 @@
+import { version as CORE_VERSION } from '../../package.json';
+import Component from './Component';
+
+/**
+ * Class that represents all data of a Plugin.
+ */
+class DefaultPluginData {
+  /**
+   * Default constructor.
+   * @param {String} props.name - Name of plugin.
+   * @param {String} props.version - Version of plugin.
+   */
+  constructor(props = {
+    name: null,
+    version: null,
+  }) {
+    /**
+     * Plugin name.
+     * @type {String}
+     */
+    this.name = props.name || null;
+    /**
+     * Plugin version.
+     * @type {String}
+     */
+    this.version = props.version || null;
+    /**
+     * All plugin components.
+     * @type {Component[]}
+     */
+    this.components = [];
+    /**
+     * All plugin links.
+     * @type {ComponentLink[]}
+     */
+    this.links = [];
+    /**
+     * All plugin definitions.
+     * @type {{components: ComponentDefinition[], links: ComponentLinkDefinition[]}}
+     */
+    this.definitions = {
+      components: [],
+      links: [],
+    };
+    /**
+     * All parser errors.
+     * @type {ParseError[]}
+     */
+    this.parseErrors = [];
+  }
+
+  /**
+   * Get version of plugin core.
+   * @returns {String} Version of plugin core.
+   */
+  get coreVersion() {
+    return CORE_VERSION;
```

Comment:
what's the difference between CORE_VERSION and this.version ?

Reply:
`CORE_VERSION` is the plugin core and this.version is specified by the plugin (terraform, ...)